### PR TITLE
Update Hyper and OpenSSL

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15868)

### CODE REVIEW

**Comment by nox** [comment]
```
Build failed, waiting for other jobs to finish...
error: failed to run custom build command for `openssl-sys v0.9.7`
process didn't exit successfully: `c:\buildbot\slave\windows-msvc-dev\build\target\debug\build\openssl-sys-ab551cd13dbf3163\build-script-build` (exit code: 101)
--- stdout
cargo:rustc-link-search=native=C:\Users\Administrator\.servo\msvc-dependencies\openssl\1.0.1t-vs2015\lib64
cargo:include=C:\Users\Administrator\.servo\msvc-dependencies\openssl\1.0.1t-vs2015\include
OPT_LEVEL = Some("0")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CC_x86_64-pc-windows-msvc = None
CC_x86_64_pc_windows_msvc = None
HOST_CC = None
CC = None
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CFLAGS_x86_64-pc-windows-msvc = None
CFLAGS_x86_64_pc_windows_msvc = None
HOST_CFLAGS = None
CFLAGS = None
PROFILE = Some("debug")
running: "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\amd64\\cl.exe" "/nologo" "/MD" "/Z7" "/I" "C:\\Users\\Administrator\\.servo\\msvc-dependencies\\openssl\\1.0.1t-vs2015\\include" "/E" "c:\\buildbot\\slave\\windows-msvc-dev\\build\\target\\debug\\build\\openssl-sys-b989cad2fc000929\\out\\expando.c"
cargo:warning=expando.c
ExitStatus(ExitStatus(0))
cargo:rustc-cfg=osslconf="OPENSSL_NO_KRB5"
cargo:rustc-cfg=osslconf="OPENSSL_NO_RFC3779"
cargo:conf=OPENSSL_NO_KRB5,OPENSSL_NO_RFC3779
cargo:rustc-cfg=ossl101
cargo:version=101

--- stderr
thread 'main' panicked at 'OpenSSL libdir at `C:\Users\Administrator\.servo\msvc-dependencies\openssl\1.0.1t-vs2015\lib64` does not contain the required files to either statically or dynamically link OpenSSL', C:\Users\Administrator\.cargo\registry\src\github.com-1ecc6299db9ec823\openssl-sys-0.9.7\build.rs:355
stack backtrace:
   0:      0x7f6083964c7 - std::panicking::default_hook::{{closure}}
                        at C:\projects\rust\src\libstd\panicking.rs:356
   1:      0x7f6083959f4 - std::panicking::default_hook
                        at C:\projects\rust\src\libstd\panicking.rs:367
   2:      0x7f608399311 - std::panicking::rust_panic_with_hook
                        at C:\projects\rust\src\libstd\panicking.rs:545
   3:      0x7f6083991c8 - std::panicking::begin_panic<collections::string::String>
                        at C:\projects\rust\src\libstd\panicking.rs:507
   4:      0x7f6083990e4 - std::panicking::begin_panic_fmt
                        at C:\projects\rust\src\libstd\panicking.rs:491
   5:      0x7f608316f68 - build_script_build::determine_mode
                        at C:\Users\Administrator\.cargo\registry\src\github.com-1ecc6299db9ec823\openssl-sys-0.9.7\build.rs:355
   6:      0x7f60831468f - build_script_build::main
                        at C:\Users\Administrator\.cargo\registry\src\github.com-1ecc6299db9ec823\openssl-sys-0.9.7\build.rs:76
   7:      0x7f60839bbc1 - panic_unwind::__rust_maybe_catch_panic
                        at C:\projects\rust\src\libpanic_unwind\lib.rs:98
   8:      0x7f60839991a - std::rt::lang_start
                        at C:\projects\rust\src\libstd\rt.rs:51
   9:      0x7f60831725b - main
  10:      0x7f6083a05d8 - __scrt_common_main_seh
                        at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:253
  11:      0x7fd5dab1841 - BaseThreadInitThunk
```

Cc @sfackler @retep998 @avadacatavra

**Comment by sfackler** [comment]
What is in C:\Users\Administrator\.servo\msvc-dependencies\openssl\1.0.1t-vs2015\lib64?

File: tests/unit/net/fetch.rs [hunk 1]
```diff
@@ -18,10 +18,11 @@ use hyper::header::{Encoding, Location, Pragma, Quality, QualityItem, SetCookie,
 use hyper::header::{Headers, Host, HttpDate, Referer as HyperReferer};
 use hyper::method::Method;
 use hyper::mime::{Mime, SubLevel, TopLevel};
-use hyper::net::Openssl;
+//use hyper::net::Openssl;
```
Comment (Ms2ger) [inline tests/unit/net/fetch.rs]: Remove pls

File: components/net/resource_thread.rs [hunk 2]
```diff
@@ -307,9 +310,11 @@ impl CoreResourceManager {
         }
     }
 
-    fn set_cookie_for_url(&mut self, request: &ServoUrl, cookie: cookie_rs::Cookie, source: CookieSource,
-                          resource_group: &ResourceGroup) {
-        if let Some(cookie) = cookie::Cookie::new_wrapped(cookie, &request, source) {
+    fn set_cookie_for_url(&mut self, request: &ServoUrl,
+                           cookie: cookie_rs::Cookie<'static>,
```
Comment (Ms2ger) [inline components/net/resource_thread.rs]: Indent

File: components/net/http_loader.rs [hunk 3]
```diff
@@ -126,42 +125,62 @@ struct NetworkHttpRequestFactory {
...
 impl NetworkHttpRequestFactory {
     fn create(&self, url: ServoUrl, method: Method, headers: Headers)
-              -> Result<HyperRequest<Fresh>, NetworkError> {
+              -> Result<HyperRequest<Fresh>, LoadError> {
         let connection = HyperRequest::with_connector(method,
                                                       url.clone().into_url().unwrap(),
                                                       &*self.connector);
 
         if let Err(HttpError::Ssl(ref error)) = connection {
             let error: &(Error + Send + 'static) = &**error;
```
Comment (Ms2ger) [inline components/net/http_loader.rs]: Still necessary?

File: components/net/http_loader.rs [hunk 4]
```diff
@@ -126,42 +125,62 @@ struct NetworkHttpRequestFactory {
...
         };
         *request.headers_mut() = headers;
 
         Ok(request)
     }
+
+}
+
+#[derive(Debug)]
+struct LoadError {
```
Comment (Ms2ger) [inline components/net/http_loader.rs]: I removed this in 87979ef65eb9eb3958f1ead564a663ba2bd117ad; any reason for it to return?

**Comment by nox** [comment]
Absolutely no idea.

Cc @edunham @larsbergstrom.

**Comment by nox** [comment]
Ugh, I'll never use Reviewable again.

**Comment by nox** [comment]
@Ms2ger Addressed your remarks.

**Comment by Ms2ger** [comment]
Thanks, fine for me.

**Comment by nox** [comment]
@metajack What's the status with your custom openssl archive?

**Comment by avadacatavra** [comment]
This is now blocked on switching to ws (#14517) per discussion here: http://logs.glob.uno/?c=mozilla%23servo#c632460

**Comment by avadacatavra** [comment]
This is now blocked on switching to ws (#14517) per discussion here: http://logs.glob.uno/?c=mozilla%23servo#c632460

**Pull request merged**